test/Microsoft.Azure.WebJobs.ServiceBus.UnitTests/EventHubTests.cs

```diff
@@ -59,47 +63,57 @@ public void GetBindingContract_MultipleDispatch_ReturnsExpectedValue()
-            input.PartitionContext = new PartitionContext();
+            input.PartitionContext = GetPartitionContext();
 
             var strategy = new EventHubTriggerBindingStrategy();
             var bindingData = strategy.GetBindingData(input);
 
             Assert.Equal(7, bindingData.Count);
             Assert.Same(input.PartitionContext, bindingData["PartitionContext"]);
-            Assert.Equal(evt.PartitionKey, bindingData["PartitionKey"]);
-            Assert.Equal(evt.Offset, bindingData["Offset"]);
-            Assert.Equal(evt.SequenceNumber, bindingData["SequenceNumber"]);
-            Assert.Equal(evt.EnqueuedTimeUtc, bindingData["EnqueuedTimeUtc"]);
+            Assert.Equal(evt.SystemProperties.PartitionKey, bindingData["PartitionKey"]);
+            Assert.Equal(evt.SystemProperties.Offset, bindingData["Offset"]);
+            Assert.Equal(evt.SystemProperties.SequenceNumber, bindingData["SequenceNumber"]);
+            Assert.Equal(evt.SystemProperties.EnqueuedTimeUtc, bindingData["EnqueuedTimeUtc"]);
             Assert.Same(evt.Properties, bindingData["Properties"]);
-            Assert.Same(evt.SystemProperties, bindingData["SystemProperties"]);
+            var sysDict = bindingData["SystemProperties"] as IDictionary<string, object>;
+            Assert.NotNull(sysDict);
+            Assert.Equal(evt.SystemProperties.PartitionKey, sysDict["PartitionKey"]);
+            Assert.Equal(evt.SystemProperties.Offset, sysDict["Offset"]);
+            Assert.Equal(evt.SystemProperties.SequenceNumber, sysDict["SequenceNumber"]);
+            Assert.Equal(evt.SystemProperties.EnqueuedTimeUtc, sysDict["EnqueuedTimeUtc"]);
         }
 
         [Fact]
         public void GetBindingData_MultipleDispatch_ReturnsExpectedValue()
         {
             var events = new EventData[3]
             {
-                new EventData(Encoding.UTF8.GetBytes("Event 1"))
-                {
-                    PartitionKey = "pk1"
-                },
-                new EventData(Encoding.UTF8.GetBytes("Event 2"))
-                {
-                    PartitionKey = "pk2"
-                },
-                new EventData(Encoding.UTF8.GetBytes("Event 3"))
-                {
-                    PartitionKey = "pk3"
-                },
+                new EventData(Encoding.UTF8.GetBytes("Event 1")),
+                new EventData(Encoding.UTF8.GetBytes("Event 2")),
+                new EventData(Encoding.UTF8.GetBytes("Event 3")),
             };
 
+            var count = 0;
+            foreach (var evt in events)
+            {
+                var sysProps = TestHelpers.New<SystemPropertiesCollection>();
+                TestHelpers.SetField(sysProps, "PartitionKey", $"pk{count++}");
+                TestHelpers.SetField(evt, "SystemProperties", sysProps);
+            }
+
             var input = new EventHubTriggerInput
             {
-                PartitionContext = new PartitionContext(),
```

Comment:
Did you mean to remove this PartitionContext? It was here so the below check would verify that we add it to the binding data. I guess the below test is succeeding because both are null now?

Reply:
Yep, passing as both are null. I'll add it back in - I may have removed it earlier before adding the test reflection to set private members